infra/prompt-flow/create-prompt-flow.sh

```diff
@@ -111,3 +116,16 @@ else
 fi
 
 rm "$flow_dag_file"
+
+echo "Setting prompt flow endpoint name in azd env"
+azd env set PROMPT_FLOW_ENDPOINT_NAME $endpoint_name
+azd env set PROMPT_FLOW_DEPLOYMENT_NAME $deployment_name
+
+echo "Setting environment variables in webapp"
+az webapp config appsettings set --name $webapp_name --resource-group $resource_group --settings PROMPT_FLOW_ENDPOINT_NAME=$endpoint_name PROMPT_FLOW_DEPLOYMENT_NAME=$deployment_name AZURE_ML_WORKSPACE_NAME=$aml_workspace
+az webapp config appsettings set --name $admin_webapp_name --resource-group $resource_group --settings PROMPT_FLOW_ENDPOINT_NAME=$endpoint_name PROMPT_FLOW_DEPLOYMENT_NAME=$deployment_name AZURE_ML_WORKSPACE_NAME=$aml_workspace
+
+echo "Assigning AzureML Data Scientist role to webapp"
+webapp_id=$(az webapp identity show --resource-group "$resource_group" --name $webapp_name --query principalId --output tsv)
+az role assignment create --role "AzureML Data Scientist" --scope "/subscriptions/${subscription_id}/resourceGroups/${resource_group}/providers/Microsoft.MachineLearningServices/workspaces/${aml_workspace}" --assignee-object-id "$webapp_id" --assignee-principal-type ServicePrincipal
+az role assignment create --role "AzureML Data Scientist" --scope "/subscriptions/${subscription_id}/resourceGroups/${resource_group}/providers/Microsoft.MachineLearningServices/workspaces/${aml_workspace}/onlineEndpoints/${endpoint_name}" --assignee-object-id "$webapp_id" --assignee-principal-type ServicePrincipal
```

Comment:
Can all of the above be added to the bicep file instead? The web apps don't reload config, so would need to be restarted after this script runs. It may require us to move the endpoint and deployment names there too, and then pass them to this script instead, but that's okay I think.

Reply:
as discussed in offline meeting, unfortunately no - as endpoint will not exist before this script is run.